wifiphisher/pywifiphisher.py

```diff
@@ -586,69 +590,73 @@ def start(self):
+                    if self.network_manager.is_interface_valid(args.apinterface, "AP"):
+                        ap_iface = args.apinterface
                 else:
-                    ap_iface = self.network_manager.get_ap_iface()
-                self.network_manager.set_ap_iface(ap_iface.get_name())
+                    ap_iface = self.network_manager.get_interface(True, False)
                 mon_iface = ap_iface
 
                 if not args.no_mac_randomization:
-                    if not args.mac_ap_interface:
-                        self.network_manager.randomize_ap_interface_mac_addr()
+                    if args.mac_ap_interface:
+                        self.network_manager.set_interface_mac(ap_iface, args.mac_ap_interface)
                     else:
-                        self.network_manager.randomize_ap_interface_mac_addr(args.mac_ap_interface)
+                        self.network_manager.set_interface_mac_random(ap_iface)
 
                 print ("[{0}+{1}] Selecting {0}{2}{1} interface for creating the "
-                       "rogue Access Point").format(G, W, ap_iface.get_name())
+                       "rogue Access Point").format(G, W, ap_iface)
                 # randomize the mac addresses
                 if not args.no_mac_randomization:
                     ap_iface.randomize_interface_mac(args.mac_ap_interface)
+
+            # make sure interfaces are not blocked
+            self.network_manager.unblock_interface(ap_iface)
+            self.network_manager.unblock_interface(mon_iface)
+
             kill_interfering_procs()
             self.network_manager.set_interface_mode(mon_iface, "monitor")
-        except (interfaces.NotEnoughInterfacesFoundError,
-                interfaces.JammingInterfaceInvalidError,
-                interfaces.ApInterfaceInvalidError,
-                interfaces.NoApInterfaceFoundError,
-                interfaces.NoMonitorInterfaceFoundError,
-                interfaces.DeauthInterfaceMacAddrInvalidError,
-                interfaces.ApInterfaceMacAddrInvalidError) as err:
-            print ("[{0}!{1}] " + str(err)).format(R, W)
+        except (interfaces.InvalidInterfaceError,
+                interfaces.InvalidInternetInterfaceError,
+                interfaces.InterfaceCantBeFoundError) as err:
+            print ("[{0}!{1}] {2}").format(R, W, err)
             time.sleep(1)
             self.stop()
 
         if not args.no_mac_randomization:
-            print ("[{0}+{1}] " + ap_iface.get_name() + ' mac address becomes '
-                     + ap_iface.get_current_mac()).format(G, W)
+            print "[{0}+{1}] {2} mac address becomes is now {3} ".format(G, W, ap_iface)
+
             if not args.nojamming:
-                print ("[{0}+{1}] " + mon_iface.get_name() + ' mac address becomes '
-                     + mon_iface.get_current_mac()).format(G, W)
+                print ("[{0}+{1}] {2} mac address becomes {3}".format(G, W, mon_iface))
```

Comment:
check this statement `self.access_point.set_interface(ap_iface.get_name())`. `ap_iface` is already a string.